packages/ui/src/components/va-button-toggle/VaButtonToggle.vue

```diff
@@ -39,45 +39,59 @@ export default defineComponent({
   },
   emits: ['update:modelValue'],
   props: {
-    options: { type: Array as PropType<any[]>, default: () => [] },
-    color: { type: String as PropType<string>, default: 'primary' },
-    textColor: { type: String as PropType<string>, default: undefined },
-    activeButtonTextColor: { type: String as PropType<string>, default: 'white' },
+    options: { type: Array, default: () => [] },
+    color: { type: String, default: 'primary' },
+    textColor: { type: String, default: undefined },
+    activeButtonTextColor: { type: String, default: 'white' },
     modelValue: { type: [String, Number] as PropType<string | number>, default: '' },
-    outline: { type: Boolean as PropType<boolean>, default: false },
-    flat: { type: Boolean as PropType<boolean>, default: false },
-    rounded: { type: Boolean as PropType<boolean>, default: true },
-    disabled: { type: Boolean as PropType<boolean>, default: false },
+    outline: { type: Boolean, default: false },
+    flat: { type: Boolean, default: false },
+    rounded: { type: Boolean, default: true },
+    disabled: { type: Boolean, default: false },
     size: {
-      type: String as PropType<string>,
+      type: String as PropType<'medium' | 'small' | 'large'>,
       default: 'medium',
       validator: (modelValue: 'medium' | 'small' | 'large') => ['medium', 'small', 'large'].includes(modelValue),
     },
 
-    toggleColor: { type: String as PropType<string>, default: '' },
-    gradient: { type: Boolean as PropType<boolean>, default: false },
+    toggleColor: { type: String, default: '' },
+    gradient: { type: Boolean, default: false },
   },
   setup (props, { emit }) {
     const { getColor } = useColors()
     const colorComputed = computed(() => getColor(props.color))
     const toggleColorComputed = computed(() => getColor(props.toggleColor))
 
-    const getButtonProps = (buttonValue: any) => {
-      if (buttonValue !== props.modelValue) { return }
+    const isFlatOrOutline = computed(() => props.outline || props.flat)
+    const color = computed(() => {
+      if (props.toggleColor) {
+        return toggleColorComputed.value
+      } else {
+        return isFlatOrOutline.value ? colorComputed.value : shiftHSLAColor(colorComputed.value, { l: -6 })
+      }
+    })
+    const textColor = computed(() => props.activeButtonTextColor || getTextColor(colorComputed.value))
```

Comment:
Look like issue with text colors fixed in this PR: https://github.com/epicmaxco/vuestic-ui/pull/1769.

Again, let's have a smaller diff.